packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -206,6 +206,28 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
 
   @override
   void insert(RenderBox child, { RenderBox after }) {
+    // `after` might be coming in from a caller that didn't realize it was in the
+    // _keepAliveBucket, in which case it needs to be removed and reinserted
+    // to get connected to its siblings and be useful as an anchor.
+    if (after != null) {
+      final SliverMultiBoxAdaptorParentData afterParentData = after.parentData;
+      final int index = afterParentData.index;
+      if (_keepAliveBucket.containsKey(index)) {
```

Comment:
rather than looking up the index in the map, i think you can just check _keptAlive on the child's parentData (then assert that it's in the map)

Reply:
(while you're at it, we should probably change _createOrObtainChild to do that too)

Reply:
_createOrObtainChild gets an index rather than a child.  Not sure that one makes sense to change to me.  But this one does make sense.